UnityProject/Assets/Scripts/MiniGames/MiniGameModules/ReflectionGolf/ReflectionGolfLevel.cs

```diff
@@ -0,0 +1,137 @@
+using System;
+using UnityEngine;
+using System.Text.RegularExpressions;
+using System.IO;
+using Logs;
+using SecureStuff;
+
+namespace MiniGames.MiniGameModules
+{
+	public class ReflectionGolfLevel
+	{
+
+		public const int DISPLAY_WIDTH = 600;
+
+		public short Width { get; private set; }
+
+		public short Height { get; private set; }
+
+		public CellData[] LevelData { get; private set; } = new CellData[0]; 
+
+		public Vector2Int GoalLocation { get; private set; }
+
+		private readonly ReflectionGolfModule miniGameModule = null;
+
+		public string LevelName { get; private set; } = "";
+
+		public ReflectionGolfLevel(Difficulty difficulty, ReflectionGolfModule _miniGameModule)
+		{
+			miniGameModule = _miniGameModule;
+
+			LevelName = ReflectionGolfLevelImporter.PickLevel(difficulty);
+			LoadLevelFromFile(LevelName);
+		}
+
+		public ReflectionGolfLevel(string levelName, ReflectionGolfModule _miniGameModule)
+		{
+			miniGameModule = _miniGameModule;
+
+			LevelName = levelName;
+			LoadLevelFromFile(levelName);
+		}
+
+		public ReflectionGolfLevel(CellData[] levelData, short width, ReflectionGolfModule _miniGameModule)
+		{
+			miniGameModule = _miniGameModule;
+
+			LevelData = levelData;
+			Width = width;
+			Height = (short)(levelData.Length / width);
+
+			InitialiseLevelValues();
+		}
+
+		public void LoadLevelFromFile(string levelName)
+		{
+			miniGameModule.ClearAllMoves();
+
+			string path = Path.Combine("MiniGamesData", "ReflectionGolf", $"{levelName}.txt");
+
+			if (AccessFile.Exists(path) == false)
+			{
+				Loggy.LogError($"MiniGames/RelfectionGolfLevel.cs at line 61. The specified file path does not exist! {path}");
```

Comment:
follow our standard for writing logs
https://unitystation.github.io/unitystation/contribution-guides/Development-Standards-Guide/#0-in-code-documentation-and-clarity